### PR TITLE
fix: raise proper exception in services pay_invoice

### PR DESCRIPTION
**Pull request by dni**
also values where checked twice

### CODE REVIEW

File: lnbits/core/services.py [hunk 1]
```diff
@@ -189,24 +189,22 @@ async def pay_invoice(
...
-        raise ValueError("Amount in invoice is too high.")
+        raise InvoiceFailure("Amount in invoice is too high.")
 
     await check_wallet_limits(wallet_id, conn, invoice.amount_msat)
 
     async with db.reuse_conn(conn) if conn else db.connect() as conn:
         temp_id = invoice.payment_hash
         internal_id = f"internal_{invoice.payment_hash}"
 
-        if invoice.amount_msat == 0:
```
Comment (motorina0) [inline lnbits/core/services.py:205]: why is this removed (more than a refactor)?

Comment (dni) [reply lnbits/core/services.py:205]: its actually already check in line 197 `not invoice.amount_msat > 0`

**Review by motorina0 — APPROVED** [review]

**Review by prusnak — APPROVED** [review]
